scipy/special/cephes/igam.c

```diff
@@ -129,9 +129,15 @@ double igam(double a, double x)
 {
     double absxma_a;
 
-    if (x < 0 || a <= 0) {
+    if (x < 0 || a < 0) {
 	sf_error("gammainc", SF_ERROR_DOMAIN, NULL);
 	return NPY_NAN;
+    } else if (a == 0) {
```

Comment:
Let's also move up the `if (x == 0)` branch from below to keep all the various-parameters-zero logic clustered together.

Reply:
done